x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/RemoteAccessServerTransportFilter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.transport;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.DestructiveOperations;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.security.SecurityContext;
+import org.elasticsearch.xpack.core.security.authc.Authentication;
+import org.elasticsearch.xpack.security.authc.AuthenticationService;
+import org.elasticsearch.xpack.security.authc.RemoteAccessAuthenticationService;
+import org.elasticsearch.xpack.security.authz.AuthorizationService;
+
+final class RemoteAccessServerTransportFilter extends ServerTransportFilter {
```

Comment:
This seems to be the right class for having an allowlist of actions. So we might need tweek the super/sub class methods a bit more to have that.

Reply:
Good catch, this is another bit that did not survive a refactoring pass. I've updated the `authenticate` method in `RemoteAccessServerTransportFilter` to run the allowlist check.